src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -612,99 +668,158 @@ def realm_rm_zone_creds(self, realm_token_b64):
 
         return (0, success_message, '')
 
-    def zone_create(self, realm_token_b64, zonegroup_name=None, zone_name=None,
-                    endpoints=None, start_radosgw=True):
-        if not realm_token_b64:
-            print('missing realm access config')
-            return False
-
-        realm_token_b = base64.b64decode(realm_token_b64)
-        realm_token_s = realm_token_b.decode('utf-8')
+    def zone_modify(self, realm_name, zonegroup_name, zone_name, endpoints, realm_token_b64):
 
-        realm_token = json.loads(realm_token_s)
-
-        access_key = realm_token['access_key']
-        secret = realm_token['secret']
+        if not realm_token_b64:
+            raise RGWAMException('missing realm access config')
+        if zone_name is None:
+            raise RGWAMException('Zone name is a mandatory parameter')
 
+        realm_token = RealmToken.from_base64_str(realm_token_b64)
+        access_key = realm_token.access_key
+        secret = realm_token.secret
         try:
-            realm_info = self.realm_op().pull(
-                realm_token['endpoint'], access_key, secret, set_default=True)
+            # We only pull the realm if we are on a secondary cluster
+            if not realm_token.is_primary and realm_token.endpoint is not None:
```

Comment:
This realm pull is not necessary. The realm pull only needs to happen when you are creating the secondary zone for the first time, not modifying it.

Reply:
AFAIR from my testing without this pull the modify zone fails (when updating the end-points of the secondary zone). I have to double check what went wrong on those cases.

Reply:
This seems strange and shouldn't be the case, let me know how double checking goes. We can do this together if you'd like.

Reply:
@alimaredia I double checked this and find out that indeed it was not needed so I just went and removed the corresponding code. Not sure why I got errors in the past when not pulling the zone during the modify.